### PR TITLE
🥊 Add thread status to mocks

### PR DESCRIPTION
**Pull request by p-sad**
So that we can move on to tasks that use it. Bad news is that `DataFields` type can no longer be used for `asForumThread` since the thread status is determined by the status's `__typename` in GQL — unless we want to write something sketchy looking like
```ts
type ThreadStatus = ForumThreadFieldsFragment['status']
export const asForumThread = (fields: DataFields<ForumThreadFieldsFragment> & { status: ThreadStatus }): ForumThread => ({
```
Personally, I think the readability of a core part of the code like these typecasts is more important than readability of storybook entries.

### CODE REVIEW

**Review by thesan — COMMENTED** [review]
Agree about the core vs the stories readability.

Just about the `ForumThread.status`. Since other types (e.g `Proposal`, `Worker`) use a union type on their `status` field instead of a string. Maybe you could export the `ThreadStatus` type you defined in `forum/helpers/storybook.ts` in `forum/types/ForumThread.ts` and use it for `ForumThread.status`. WDYT ?

File: packages/ui/src/forum/types/ForumThread.ts [hunk 1]
```diff
@@ -11,6 +10,7 @@ export interface ForumThread {
   categoryId: string
   tags: ForumThreadTag[]
   visiblePostsCount: number
+  status: string
```
Comment (thesan) [inline packages/ui/src/forum/types/ForumThread.ts:13]: I mean for this `status` field

**Review by thesan — COMMENTED** [review]
Agree about the core vs the stories readability.

Just about the ForumThread.status. Since other types (e.g Proposal, Worker) use a union type on their status field instead of a string. Maybe you could export the ThreadStatus type you defined in forum/helpers/storybook.ts in forum/types/ForumThread.ts and use it for ForumThread.status. WDYT ?

File: packages/ui/src/forum/types/ForumThread.ts (hunk 1 above)
Comment (p-sad) [reply packages/ui/src/forum/types/ForumThread.ts]: You're absolutely right, I think I put it in as a string for the meantime at first and then just forgot

**Review by thesan — APPROVED** [review]

**Pull request merged**